qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/IndexingIT.java

```diff
@@ -275,6 +279,81 @@ public void testUpdateSnapshotStatus() throws Exception {
         request.setJsonEntity("{\"indices\": \"" + index + "\"}");
     }
 
+    public void testResyncFromNewNodeToOldNode() throws Exception {
```

Comment:
I wrote a rolling upgrade test but was unable to make it fail. I'll give it another shot today.

Reply:
@bleskes The reason that the rolling-upgrade did not fail because (1) primary-replica resync only operations after the global checkpoint, (2) we sync the global checkpoint for every 30 seconds. These make the resync a noop.
Since the newly added test is complicated, should we remove it and let the CCR upgrade-tests verify this BWC issue? WDYT?

Reply:
Thanks for checking. I think it's good to have a rolling upgrade test that properly tests all aspects of primary promotions. Maybe we should slow down the GCP sync through a settings (and add concurrent indexing). This may get tricky too but I think we should explore it. @ywelsch any thoughts on this?